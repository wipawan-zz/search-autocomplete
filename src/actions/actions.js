// action types
const CHANGE_QUERY = "CHANGE_QUERY";
const FETCH_REPOS = "FETCH_REPOS";
const CLEAR_AUTOCOMPLETE = 'CLEAR_AUTOCOMPLETE';
const SELECT_ITEM = 'SELECT_ITEM';

export function changeQuery(query) {
  return {
    type: CHANGE_QUERY,
    query: query
  };
}

export function fetchRepos() {
  return {
    type: FETCH_REPOS
  };
}

export function clearSearch() {
  return {
    type: CLEAR_AUTOCOMPLETE
  }
}

export function selectItem(item) {
  return {
    type: SELECT_ITEM,
    selectedItem: item
  }
}