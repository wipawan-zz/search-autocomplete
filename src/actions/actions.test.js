import configureStore from 'redux-mock-store';

// Actions to be tested
import * as Actions from './actions';

const mockStore = configureStore();
const store = mockStore();

describe('actions', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('dispatches correct query when input changes', () => {
    store.dispatch(Actions.changeQuery('test'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches correct action.type to fetch repositories', () => {
    store.dispatch(Actions.fetchRepos());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches correct action.type to clear search input', () => {
    store.dispatch(Actions.clearSearch());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches correct item when item is selected', () => {
    store.dispatch(Actions.selectItem({
      "name": "test"
    }));
    expect(store.getActions()).toMatchSnapshot();
  });
});