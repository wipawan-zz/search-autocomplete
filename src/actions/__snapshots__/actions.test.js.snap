// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions dispatches correct action.type to clear search input 1`] = `
Array [
  Object {
    "type": "CLEAR_AUTOCOMPLETE",
  },
]
`;

exports[`actions dispatches correct action.type to fetch repositories 1`] = `
Array [
  Object {
    "type": "FETCH_REPOS",
  },
]
`;

exports[`actions dispatches correct item when item is selected 1`] = `
Array [
  Object {
    "selectedItem": Object {
      "name": "test",
    },
    "type": "SELECT_ITEM",
  },
]
`;

exports[`actions dispatches correct query when input changes 1`] = `
Array [
  Object {
    "query": "test",
    "type": "CHANGE_QUERY",
  },
]
`;
